AFQ/api.py

```diff
@@ -71,135 +72,206 @@ def do_preprocessing():
+                    for hemi in ['_R', '_L']:
+                        if (templates.get(name + '_roi1' + hemi)
+                                and templates.get(name + '_roi2' + hemi)
+                                and templates.get(name + hemi + '_prob_map')):
+                            afq_bundles[name + hemi] = {
+                                'ROIs': [templates[name + '_roi1' + hemi],
+                                         templates[name + '_roi2' + hemi]],
+                                'rules': [True, True],
+                                'prob_map': templates[
+                                    name + hemi + '_prob_map'],
+                                'cross_midline': False,
+                                'uid': uid}
+                        else:
+                            self.logger.warning(
+                                f"{name} is not in AFQ templates")
+
+                        uid += 1
+        elif self.seg_algo.startswith("reco"):
+            if self.seg_algo.endswith("80"):
+                bundle_dict = afd.read_hcp_atlas(80)
             else:
-                for hemi in ["_R", "_L"]:
-                    afq_bundles[name + hemi] = bundle_dict[name + hemi]
-                    afq_bundles[name + hemi]['uid'] = uid
+                bundle_dict = afd.read_hcp_atlas(16)
+
+            afq_bundles = {}
+            uid = 1
+            afq_bundles["whole_brain"] = bundle_dict["whole_brain"]
+            for name in self.bundle_names:
+                if name in RECO_UNIQUE:
+                    afq_bundles[name] = bundle_dict[name]
+                    afq_bundles[name]['uid'] = uid
                     uid += 1
-    else:
-        raise ValueError("Input: %s is not a valid input`seg_algo`" % seg_algo)
+                else:
+                    for hemi in ["_R", "_L"]:
+                        afq_bundles[name + hemi] = bundle_dict[name + hemi]
+                        afq_bundles[name + hemi]['uid'] = uid
+                        uid += 1
+        else:
+            raise ValueError(
+                "Input: %s is not a valid input`seg_algo`" % self.seg_algo)
+        self.bundle_dict_unresampled = afq_bundles
+
+    def _gen_resampled_dict(self):
+        if self.bundle_dict_unresampled is None:
+            self._list_to_dict()
+
+        if self.seg_algo == "afq" and self.resample_to:
+            self.bundle_dict_resampled = self.bundle_dict_unresampled.copy()
+            for bundle in self.bundle_dict_unresampled:
+                rois = self.bundle_dict_unresampled[bundle]['ROIs']
+                for ii, roi in enumerate(rois):
+                    self.bundle_dict_resampled[bundle]['ROIs'][ii] =\
+                        afd.read_resample_roi(
+                            roi, resample_to=self.resample_to)
+        else:
+            self.bundle_dict_resampled = self.bundle_dict_unresampled
+
+        self._dict = self.bundle_dict_resampled
+
+    def __getattribute__(self, attr):
```

Comment:
I bet this could be made to be even lazier to set only the specific bundle each time, by passing the key into `_gen_resampled_dict`. But maybe that's not crucial 😄

Reply:
I like this idea!